logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/ConversationEventReceiver.kt

```diff
@@ -58,7 +59,12 @@ class ConversationEventReceiver(
                     )
                     //TODO Multiplatform logging
                     println("Message received: $message")
-                    messageRepository.persistMessage(message)
+                    when (message.content) {
+                        is MessageContent.Text -> messageRepository.persistMessage(message)
+                        is MessageContent.DeleteMessage -> messageRepository.softDeleteMessage(messageUuid = message.content.messageId)
```

Comment:
We are missing a check that DeleteMessage request came from the original sender of the message.

Reply:
I'm not sure that I get your point, please check the update code I think I fixed it.

Reply:
So the problem is this:
What prevents User B from sending a DeleteMessage which deletes a message sent by User A. AFAIK nothing prevents this from happening right now so we need to have check somewhere which validate that DeleteMessage was sent by the original sender of the message which should be deleted.

Reply:
Okkk! Thanks for the clarification, I didn't know that. I'll handle it.

Reply:
Just fixed it. Please check again.